### ISSUE TITLE
404 template not showing when expected

### ISSUE THREAD

**Issue by fbartels** [shaarli/Shaarli#827]
Hi,

I am currently in the process of moving my sharrli to a different domain and encountered a weird behaviour of the 404 template.

Current setup:
I have a link shortener running under my.domain and my shaarli running as links.my.domain. My idea was to make the link shortener obsolete and put shaarli directly under my.domain. This will result in shaarli being called with urls like my.domain/abc.

But when I now call my.domain/abc I get a 404 page from my webserver instead of the one defined in shaarli. I also did some more digging and calling my.domain/?123 will just show the shaarli homepage and the only way to trigger the custom 404 page in shaarli is to call my.domain/?123456.

To recap/steps to reproduce:

shaarli is running under my.domain
user is openen the page /123
Result: Webserver gives 404 page
Expected result: Shaarli gives 404 page

shaarli is running under my.domain
user is openen the page /?123
Result: I see the normal shaarli page
Expected result: Shaarli should display a 404 page

It seems the url to trigger the shaarli 404 page has to be exactly six characters long, e.g. /?123456 gives a 404, but /?12345 and /?1234567 don't.

**Comment by ArthurHoaro** [comment]
Hi @fbartels,

This is the expected behaviour, due to current limitations.

> user is openen the page /123
Result: Webserver gives 404 page
Expected result: Shaarli gives 404 page

Shaarli doesn't use URL rewriting for now (this is planned), so your web server is looking for the file `123` and can't find it, so it renders its 404 page.

> user is openen the page /?123
Result: I see the normal shaarli page
Expected result: Shaarli should display a 404 page

Permalinks use a URL *key* parameter as an ID instead of a value associated with a fixed key, so our only way to detect a permalink was to detect a 6 chars long string key parameter. However, with the new ID system added in `v0.8`, this may be improved.

I suggest that we let this issue open as a reference.

**Comment by fbartels** [comment]
Hi @ArthurHoaro ,

thanks for your feedback and explanation. As a workaround I've added `ErrorDocument 404 /?123456` to my vhost, which now redirects at least the 404 case.
